packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1077,15 +1077,16 @@ class TextSelectionGestureDetectorBuilder {
   ///    this callback.
   @protected
   void onDragSelectionStart(DragStartDetails details) {
-    final PointerDeviceKind? kind = details.kind;
-    _shouldShowSelectionToolbar = kind == null
-      || kind == PointerDeviceKind.touch
-      || kind == PointerDeviceKind.stylus;
-
-    renderEditable.selectPositionAt(
-      from: details.globalPosition,
-      cause: SelectionChangedCause.drag,
-    );
+    if (delegate.selectionEnabled) {
```

Comment:
These could return early if a smaller diff is preferred. All the other input handlers around had it like this, so I chose to keep it consistent with the surroundings.

Reply:
Actually, returning early is preferred, in both cases.